grease/src/Grease/Refine.hs

```diff
@@ -214,12 +215,20 @@ combiner = C.Combiner $ \mr1 mr2 -> do
   r1 <- mr1
   case C.subgoalResult r1 of
     -- can't refine further, no matter what the other result would be
-    ProveBug{} -> pure r1
     ProveCantRefine{} -> pure r1
     -- if we find a refinement, don't bother with other goals
     ProveRefine{} -> pure r1
     -- if this goal succeeded, continue with the others
     ProveSuccess -> mr2
+    -- a bug is only reachable if there are no other errors
+    ProveBug{} -> do
```

Comment:
Should we be returning r1 in the case below here for ProveBug under ProveNoHeuristic wouldnt that mean we end up in a case where we have some unsolved goals and then we return a bug anyways?